### ISSUE TITLE
Support for ANSI color codes

### ISSUE THREAD

**Issue by nefarius** [kubetail-org/kubetail#4]

**Comment by nefarius** [comment]
For example this is how it gets interpreted in Windows Terminal:

![image](https://github.com/kubetail-org/kubetail/assets/286631/28ada03e-498b-4e42-914e-bce6d709869c)

**Comment by nodesocket** [comment]
Maybe this one? https://github.com/drudru/ansi_up

**Comment by amorey** [comment]
I just published a new version (0.1.5) that adds support for ANSI color codes. To maintain the visual connection to the container color, lines that have ANSI codes will get prefaced with a color-coded dot. Check it out and let me know what you think:

```sh
helm repo update kubetail
helm upgrade kubetail kubetail/kubetail --namespace kubetail
```

**Comment by nefarius** [comment]
Hm, just tested it, do I need to change some setting or similar, it is not happy about what my services spit out:

![image](https://github.com/kubetail-org/kubetail/assets/286631/2ca38176-6217-4395-9c82-34080070e536)

Update went through:

```bash
helm ls -a -n kubetail
NAME            NAMESPACE       REVISION        UPDATED                                 STATUS          CHART           APP VERSION
kubetail        kubetail        7               2024-02-20 12:07:00.3693052 +0100 CET   deployed        kubetail-0.2.3  0.1.5
```

**Comment by amorey** [comment]
Hmm... sorry about that. You shouldn't have to change anything. It looks like it detected ANSI but the HTML conversion didn't go through properly. Can you send me a (sanitized) example of one of your original log lines? I'll do more testing on my end too.

**Comment by nefarius** [comment]
I've attached an anonymized log snippet produced via:

```PowerShell
kubectl logs -lapp=my-deployment --all-containers=true | Out-File -Encoding utf8 -FilePath d:\colored-logs.txt
```
So I hope it got preserved as kubetail would receive it as well: [colored-logs.txt](https://github.com/kubetail-org/kubetail/files/14348374/colored-logs.txt)

**Comment by amorey** [comment]
Thanks! The encoding was preserved. Will take a look.

**Comment by amorey** [comment]
Ok, just pushed a new version with a bug fix (0.1.6). Here's an example on the demo site:
https://demo.kubetail.com/console?source=deployments%2Fkubetail-demo%2Floggen-ansi

Let me know if that fixes the problem.

**Comment by nefarius** [comment]
Well, partial good news! We have readable text back, however now no colors at all 😅 Guess this is not how it's supposed to be?

![image](https://github.com/kubetail-org/kubetail/assets/286631/a4ad15ac-0a49-4c12-a40e-3cb2c919d12b)

**Comment by amorey** [comment]
The latest release (0.1.7) should help but there's still work to be done. I switched to the `ansi_up` package which supports more ANSI codes but it assumes a dark background so for now I made the background of the message column dark when ANSI encodings are found:

![Kubetail___Log_Viewer](https://github.com/kubetail-org/kubetail/assets/75881/58205094-19a0-4737-9256-e2f1602f6bd0)

https://demo.kubetail.com/console?source=deployments%2Fkubetail-demo%2Floggen-ansi

Let me know if the ANSI encodings are being handled correctly on your end. I'll create another issue to track further improvements to ANSI handling.
